es-app/src/guis/GuiCollectionSystemsOptions.cpp

```diff
@@ -184,11 +190,15 @@ void GuiCollectionSystemsOptions::updateSettings(std::string newAutoSettings, st
 	Settings::getInstance()->setString("CollectionSystemsCustom", newCustomSettings);
 	Settings::getInstance()->setBool("SortAllSystems", sortAllSystemsSwitch->getState());
 	Settings::getInstance()->setBool("UseCustomCollectionsSystem", bundleCustomCollections->getState());
+	Settings::getInstance()->setBool("CollectionShowSystemInfo", toggleSystemNameInCollections->getState());
 	Settings::getInstance()->saveFile();
+
+	CollectionSystemManager::get()->toggleDisplayNameChange(true);
```

Comment:
So, unsure about why this needs to be in two steps (I imagine there's a strong reason for it), as I imagine this is really just for setting the bit "dirty" to update the necessary things.

Still, the main comment (assuming it can't be done in a single step - say, "setDisplayNameDirty") is that this isn't really a "toggle" as much as it is a "setDisplayNameChange" as we need to pass it the actual state :) I know it's a small nitpick, but if this is effectively the way to go, I'd rename the method to "set" rather than toggle. Other than that, looks good and tested!

Reply:
The reason for the 'toggle' is indeed to force a name change for _mCollectionFileName [here](https://github.com/RetroPie/EmulationStation/pull/571/files#diff-be83cc0bb631caf580f16dbb58693818L348). Alternatively, I think the list of Collection FileData(s) can be walked and the _mCollectionFileName_ updated for each entry - I'm now thinking if this method wouldn't be more efficient - the 'toggle' forces the re-calculation and update of all collections.

No objection for the naming, can be changed.

Reply:
What about doing something like the following?  Then you wouldn't even need the toggleDisplayNameChange method.

```cpp
const std::string& CollectionFileData::getName()
{
	if (mDirty) {
		mCollectionFileName  = Utils::String::removeParenthesis(mSourceFileData->metadata.get("name"));
		mCollectionFileName += " [" + Utils::String::toUpper(mSourceFileData->getSystem()->getName()) + "]";
		mDirty = false;
	}

	if (Settings::getInstance()->getBool("CollectionShowSystemInfo"))
		return mCollectionFileName;
	return mSourceFileData->metadata.get("name");
}


```

Reply:
Actually, and this is maybe out of scope for this PR, but we could move the `if (mDirty)` name regeneration logic up to the refreshMetadata method and get rid of the `mDirty`.

Reply:
@jrassa yes, that works - less code. I guess I overcomplicated things by trying not to trigger the re-evaluation of the _mCollectionFileName_.
The one thing that's changing from the default the removal of _removeParanthesis_ - it potentially changes the name (and ordering) of the games when removing the system name. I guess that's fine, since _removeParanthesis_ was probably meant to remove any name fragments that would have been mistaken for system name or just cause confusion when looking at the list and have both parentheses from game name and system name.

I'll re-test and re-push the changes.